architecture/2024-08-applicationdeployment-controller.md

```diff
@@ -0,0 +1,370 @@
+# ApplicationDeployment Controller
+
+* **Author**: Will Smith (@willdavsmith)
+
+## Overview
+
+<!--
+Provide a succinct high-level description of the component or feature and 
+where/how it fits in the big picture. The overview should be one to three 
+paragraphs long and should be understandable by someone outside the Radius
+team. Do not provide the design details in this, section - there is a
+dedicated section for that later in the document.
+-->
+
+Currently, there is little support for deploying Radius resources without the use of the Radius CLI. This complicates future scenarios (such as support for GitOps) because GitOps and other tools that we want to integrate with are unable to deploy Radius resources (as defined in Bicep manifests) directly. Today, Radius is primarily deployed onto Kubernetes clusters, and naturally most of the existing Kubernetes ecosystem only works with Kubernetes resources. This design document proposes updates to Radius will allow users to deploy Radius resources by authoring Kubernetes CRDs.
+
+## Terms and definitions
+
+<!--
+Include any terms, definitions, or acronyms that are used in
+this design document to assist the reader. They may or may not
+be part of the user-facing experience once implemented, and can
+be specific to this design context.
+-->
+
+**CRD (Custom Resource Definition)**: A Kubernetes resource that allows users to define their own resource types.
+
+**GitOps** A way to do Kubernetes cluster management and application delivery. It works by using Git as a single source of truth for declarative infrastructure and applications.
+
+## Objectives
+
+<!--
+Describe goals/non-goals and user-scenario of this feature to understand
+the end-user goals.
+* If the feature shares the same objectives of the existing design, link
+  to the existing doc and section rather than repeat the same context.
+* If the feature has a scenario, UX, or other product feature design doc,
+  link it here and summarize the important parts.
+-->
+
+### Goals
+
+<!--
+Describe goals to define why we are doing this work, how we will make
+priority decisions, and how we will determine success.
+-->
+
+**Goal: Radius resources can be deployed by authoring a Kubernetes CRD**
```

Comment:
What about non-Radius usecases for Bicep? in-scope? out of scope?

Reply:
What about non-application scenarios - like updating the configuration of an environment?